### DISCUSSION TITLE
CR Touch not working after custom firmware update

### DISCUSSION THREAD

**Discussion by brunohirt** [mriscoc/Ender3V2S1#397] [Q&A]
Hello! 
I used the firmware configurator to make a custom F4 chip firmware, I download the latest release source code, then I selected:
1- Ender3S1
2- 301F4
3- BLT  with UBL disabled  (my  S1 Pro has a CR touch, but there's no CR touch option here)
4- T13 (Mine is a S1 Pro)

In VsCode, I compiled it to a .bin file, and tried it out, but when turning on the printer, CR touch light remains red, not purple, and it doesn't work after that. So I found out that MRiscoC made a F4+T13 firmware on Septemebr 17th (basically the same firmware I tried to do manually), I downloaded it and tried it out, CR touch is now working.

So, what I'm doing wrong on the firmware code that is not making my CR Touch work? Can anybody help me?

Thank you very much!

**Comment by brunohirt** [comment]
Got it to work not using the configurator, but directly changing the code from the soruce code for F4 chip, but now I'm having a strange issue, like 50% of the times I turn the printer on it goes purple and works, 50% it stays red and doesn't. Tried to reflash the firmware, it continues like this. Not a big deal right now, as I just need to turn it off and on again 1-3 times and it'll work.

What can it be?

**Reply by mriscoc** [reply]
Could you post your configuration files?

**Reply by brunohirt** [reply]
https://we.tl/t-2YE0RNZg4b

Both configuration.h and _adv.h are here.
Basically I used a clean F4 source code, disabled ARC (to free firmware memory), enabled LA and changed to T13 settings, directly in VsCode. Did not change anything else. But will try to reenable ARC and disable those features you wrote some minutes ago on my other post here.

**Reply by brunohirt** [reply]
Curious thing, if the SD card with the firmware file used to flash it the last time is in the printer, CR touch always works first try. If I turn the printer on without the SD card containing the firmware (.gcode or other files don't work) it sometimes go purple and works, but mostly don't.

**Reply by mriscoc** [reply]
Enable `Media Update` in Advanced Settings menu.

**Reply by brunohirt** [reply]
No luck again. Enabling Media Update didn't change the behaviour, it still doesn't work most of the time while without the SD card containing the last firmware file used.
